tests/components/dhcp/test_init.py

```diff
@@ -199,6 +200,52 @@ async def test_dhcp_match_hostname_and_macaddress(hass: HomeAssistant) -> None:
     )
 
 
+async def test_dhcp_match_hostname_and_macaddress_specific_interfaces(
+    hass: HomeAssistant,
+) -> None:
+    """Test matching based on hostname and macaddress."""
+    integration_matchers = dhcp.async_index_integration_matchers(
+        [{"domain": "mock-domain", "hostname": "connect", "macaddress": "B8B7F1*"}]
+    )
+
+    def _generate_mock_adapters():
+        mock_eth0 = Mock(spec=ifaddr.Adapter)
+        mock_eth0.nice_name = "eth0"
+        mock_eth0.ips = [ifaddr.IP("192.168.0.1", 24, "eth0")]
+        mock_eth0.index = 1
+        mock_eth1 = Mock(spec=ifaddr.Adapter)
+        mock_eth1.nice_name = "eth1"
+        mock_eth1.ips = [ifaddr.IP("192.168.1.1", 24, "eth1")]
+        mock_eth1.index = 2
+        return [mock_eth0, mock_eth1]
+
+    packet = Ether(RAW_DHCP_REQUEST)
+
+    with patch(
+        "homeassistant.components.network.util.ifaddr.get_adapters",
```

Comment:
```suggestion
        "homeassistant.components.dhcp.network.async_get_adapters",
```

We want to patch the public API of the network integration since these tests are for dhcp and the code uses the public API of network

Reply:
Ah, I thought that one was public too, thanks.